pilot/pkg/bootstrap/server_test.go

```diff
@@ -767,3 +762,22 @@ func TestGetDNSNames(t *testing.T) {
 		})
 	}
 }
+
+func setupTestSigner(c kube.Client) {
+	csrs := krt.NewInformer[*cert.CertificateSigningRequest](c)
+	krt.NewCollection[*cert.CertificateSigningRequest, struct{}](csrs,
+		func(ctx krt.HandlerContext, csr *cert.CertificateSigningRequest) *struct{} {
+			if len(csr.Status.Certificate) > 0 {
+				return nil
+			}
+			for _, cond := range csr.Status.Conditions {
+				if cond.Type == cert.CertificateApproved && cond.Status == corev1.ConditionTrue {
+					// for this test only, sign the request when it is approved
+					csr.Status.Certificate = testcerts.ServerCert
+					c.Kube().CertificatesV1().CertificateSigningRequests().UpdateStatus(context.Background(), csr, metav1.UpdateOptions{})
+					return nil
+				}
+			}
+			return nil
+		})
+}
```

Comment:
```suggestion
func setupTestSigner(c kube.Client) {
	csrs := krt.NewInformer[*cert.CertificateSigningRequest](c)
	updates := krt.NewCollection[*cert.CertificateSigningRequest, cert.CertificateSigningRequest](csrs,
		func(ctx krt.HandlerContext, csr *cert.CertificateSigningRequest) *cert.CertificateSigningRequest {
			if len(csr.Status.Certificate) > 0 {
				return nil
			}
			for _, cond := range csr.Status.Conditions {
				if cond.Type == cert.CertificateApproved && cond.Status == corev1.ConditionTrue {
					// for this test only, sign the request when it is approved
					res := csr.DeepCopy()
					res.Status.Certificate = testcerts.ServerCert
					return res
				}
			}
			return nil
		})
	updates.Register(func(o krt.Event[cert.CertificateSigningRequest]) {
		if o.Event == controllers.EventDelete {
			return
		}
		c.Kube().CertificatesV1().CertificateSigningRequests().UpdateStatus(context.Background(), o.New, metav1.UpdateOptions{})
	})
}
```

This avoids mutations in the transformation